src/utilities/export-pdf-helper.ts

```diff
@@ -0,0 +1,242 @@
+import User from '../database/models/user';
+import CourseTopicContent from '../database/models/course-topic-content';
+import CourseWWTopicQuestion from '../database/models/course-ww-topic-question';
+import StudentGrade from '../database/models/student-grade';
+import logger from './logger';
+import StudentWorkbook from '../database/models/student-workbook';
+import StudentGradeInstance from '../database/models/student-grade-instance';
+import ProblemAttachment from '../database/models/problem-attachment';
+import rendererHelper from './renderer-helper';
+import Role from '../features/permissions/roles';
+import configurations from '../configurations';
+import _ = require('lodash');
+import axios from 'axios';
+import NotFoundError from '../exceptions/not-found-error';
+import moment = require('moment');
+import { asyncForOwn } from '../extensions/object-extension';
+import urljoin = require('url-join');
+
+interface BulkPdfExport {
+    firstName: string;
+    lastName: string;
+    topic: {
+        name: string;
+        id: number;
+    };
+    professorUUID: string;
+    problems: {
+        number: number;
+        srcdoc: string;
+    }[];
+}
+
+// topic from db
+type FROM_DB = {
+    id: number;
+    name: string;
+    questions: {
+        id: number;
+        problemNumber: number;
+        webworkQuestionPath: string;
+        grades: {
+            id: number;
+            lastInfluencingCreditedAttemptId: number;
+            lastInfluencingAttemptId: number;
+            user: User;
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            rendererData: any;
+            webworkQuestionPath?: string;
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            problemAttachments: any;
+            randomSeed: number;
+        }[];
+    }[];
+}
+
+export default class ExportPDFHelper {
+    bulkExportAxios = axios.create({
+        baseURL: configurations.bulkPdfExport.baseUrl,
+    });
+
+    shouldStartNewExport = (topic: CourseTopicContent): boolean => {
+        // If the time is within 5 minutes, do not allow a new export
+        if (!_.isNil(topic.lastExported) && moment().isBefore(moment(topic.lastExported).add(5, 'minutes'))) {
+            return false;
+        }
+
+        return true;
+    }
+    
+    start = async ({topic, professorUUID}: {topic: CourseTopicContent; professorUUID: string}): Promise<void> => {
+        topic.lastExported = new Date();
+        topic.exportUrl = null;
+        await topic.save();
```

Comment:
I prefer for these microservice helpers to not make calls the to the database

Reason being is that I think of them as sdks

If we ever wanted to use this in another application or external script we wouldn't be able to